mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -175,6 +181,62 @@ impl PallasChainObserver {
         Ok(utxo)
     }
 
+    /// Fetches the current stake distribution using the provided `statequery` client.
+    async fn do_stake_snapshots_state_query(
+        &self,
+        statequery: &mut Client,
+    ) -> StdResult<StakeSnapshot> {
+        statequery
+            .acquire(None)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to acquire statequery")?;
+
+        let era = queries_v16::get_current_era(statequery)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to get current era")?;
+
+        let state_snapshot = queries_v16::get_stake_snapshots(statequery, era, BTreeSet::new())
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to get stake snapshot")?;
+
+        Ok(state_snapshot)
+    }
+
+    fn get_stake_pool_hash(
+        &self,
+        key: &Bytes,
+        stakes: &Stakes,
+    ) -> Result<(String, u64), ChainObserverError> {
+        let pool_hash = bech32::encode("pool", key.to_base32(), Variant::Bech32)
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to encode stake pool hash")?;
+
+        Ok((pool_hash, stakes.snapshot_mark_pool))
+    }
+
+    async fn get_stake_distribution_snapshot(
+        &self,
+        client: &mut NodeClient,
+    ) -> Result<Option<StakeDistribution>, ChainObserverError> {
+        let statequery = client.statequery();
+
+        let stake_snapshot = self.do_stake_snapshots_state_query(statequery).await?;
+
+        let mut stake_distribution = StakeDistribution::new();
+
+        for (key, stakes) in stake_snapshot.snapshots.stake_snapshots.iter() {
+            if stakes.snapshot_mark_pool > 0 {
+                let (pool_hash, stake) = self.get_stake_pool_hash(key, stakes)?;
+                stake_distribution.insert(pool_hash, stake);
+            }
+        }
+
+        Ok(Some(stake_distribution))
```

Comment:
why are we passing the `stakes` value to the `get_stake_pool_hash` only to get the `snapshot_mark_pool` value back?

wouldn't it make more sense to just do:

```
let pool_hash = self.get_stake_pool_hash(key)?;
stake_distribution.insert(pool_hash, stakes.snapshot_mark_pool);
```

Reply:
thank you!